tests/test_settings.py

```diff
@@ -1808,6 +1811,21 @@ def test_builtins_settings_source_repr():
     )
 
 
+@pytest.mark.parametrize(
+    'cls, expected',
+    (
+        (JsonConfigSettingsSource, 'JsonConfigSettingsSource(json_file=config.json)'),
+        (TomlConfigSettingsSource, 'TomlConfigSettingsSource(toml_file=config.toml)'),
+        (YamlConfigSettingsSource, 'YamlConfigSettingsSource(yaml_file=config.yaml)'),
+    ),
+)
+def test_configfile_settings_source_repr(cls: Type[PydanticBaseSettingsSource], expected: str) -> None:
+    ext = cls.__name__[:4].lower()
+    path = Path(f'config.{ext}')
+    source = cls(BaseSettings(), path)
+    assert repr(source) == expected
+
+
```

Comment:
Please separate the tests and move them into 

- test_source_json.py
- test_source_toml.py
- test_source_yaml.py